features/contract-state/Totals.tsx

```diff
@@ -87,10 +89,29 @@ const Totals = () => {
     prettyTotalTokens: string = defaultMissingDataDisplay,
     prettyCollSymbol: string = "",
     prettyTokenSymbol: string = "",
-    getExchangeLinkCollateral: string = "https://app.uniswap.org/#/swap",
-    getExchangeLinkToken: string = "https://app.uniswap.org/#/swap",
+    getExchangeLinkCollateral: string = "",
+    getExchangeLinkToken: string = "",
     exchangeName: string = "Uniswap"
   ) {
+    const addTokenToMetamask = async () => {
```

Comment:
You should add `provider !== null` to the early `if` statement checking for null values, and then declare `addTokenToMetamask` within that if statement.

Then you should pass `addTokenToMetamask` as a function into `renderComponent`. This would be consistent with the repo style

Reply:
I _can_ do that but this feels like it deviates from the pattern tbh. The `renderComponent` logic is not used to pas around functions like this in other places.

Reply:
I implemented the changes you suggested. take a look and let me know if this is what you meant